pkgs/tools/security/ghidra/build-extension.nix

```diff
@@ -0,0 +1,78 @@
+{ lib
+, stdenv
+, unzip
+, jdk
+, gradle
+, ghidra
```

Comment:
another instance of broken recursion structure, i think, actually 🙃

the stuff should probably be re-callPackage'd inside the passthru of ghidra, maybe? I am not super confident that this is the ideal fix, but this will also possibly confuse someone in the future.

or we can not expose it through `passthru`, which would make people expect that it would not be affected by `overrideAttrs`. just, as long as we expose it through `ghidra`, i think that people will expect `overrideAttrs` to work.

Reply:
I really like the idea to expose it through `ghidra` because i think that makes this feature much more discoverable, but maybe it's easier to just expose it as a completely differentPackage, that takes `ghidra` as an argument.
Am I correct in the assumption, that this would greatly simplify everything?

Is there a way to just re-callPackage this in the `passthru` and have it be picked up everywhere afterwards automatically or would we also need to pass these modified `buildGhidraExtension` and `buildGhidraScripts`  explicitly throughout the entire callchain?

Reply:
yes, if we use `.override` the ones we put through passthru with the same finalPackage arg as we used for callPackage, that should just fix it, i believe?

Reply:
I'll have a stab at this tomorrow. Thank you once again for your willingness to help and teach!

Reply:
Okay, I was able to confirm (by checking `ghidra.pname`) that the correct package arrives in the extension builds and the builder.